src/Aspire.Hosting.Docker/DockerComposePublishingContext.cs

```diff
@@ -0,0 +1,521 @@
+        private readonly Dictionary<string, string> _allocatableParameters = [];
+        private readonly Dictionary<string, EndpointMapping> _endpointMapping = [];
+
+        public IResource Resource => resource;
+
+        public List<ComposeEnvironmentVariable> EnvironmentVariables { get; } = [];
+
+        public List<ComposeCommand> Args { get; } = [];
+
+        public Dictionary<string, object> Parameters { get; } = [];
+
+        public List<ComposeVolume> Volumes { get; } = [];
+
+        public ComposeService BuildComposeService()
+        {
+            if (!TryGetContainerImageName(resource, out var containerImageName))
+            {
+                composePublishingContext.Logger.FailedToGetContainerImage(resource.Name);
+            }
+
+            var composeService = new ComposeService(resource.Name.ToLowerInvariant(), composePublishingContext.Options.ExistingNetworkName);
+
+            SetEntryPoint(composeService);
+            AddEnvironmentVariablesAndCommandLineArgs(composeService);
+            AddPorts(composeService);
+            SetContainerImage(containerImageName, composeService);
+
+            return composeService;
+        }
+
+        private void AddPorts(ComposeService composeService)
+        {
+            if (_endpointMapping.Count == 0)
+            {
+                return;
+            }
+
+            foreach (var (_, mapping) in _endpointMapping)
+            {
+                var port = mapping.Port.ToString(CultureInfo.InvariantCulture);
+                var targetPort = mapping.TargetPort?.ToString(CultureInfo.InvariantCulture) ?? port;
+
+                composeService.AddPort($"{port}:{targetPort}");
+            }
+        }
+
+        private static void SetContainerImage(string? containerImageName, ComposeService composeService)
+        {
+            if (containerImageName is not null)
+            {
+                composeService.WithImage(containerImageName);
+            }
+        }
+
+        private static bool TryGetContainerImageName(IResource resource, out string? containerImageName)
+        {
+            // If the resource has a Dockerfile build annotation, we don't have the image name
+            // it will come as a parameter
+            if (resource.TryGetLastAnnotation<DockerfileBuildAnnotation>(out _))
+            {
+                containerImageName = null;
+                return false;
+            }
+
+            return resource.TryGetContainerImageName(out containerImageName);
+        }
+
+        public async Task ProcessResourceAsync(DistributedApplicationExecutionContext executionContext, CancellationToken cancellationToken)
+        {
+            ProcessEndpoints();
+            ProcessVolumes();
+
+            await ProcessEnvironmentAsync(executionContext, cancellationToken).ConfigureAwait(false);
+            await ProcessArgumentsAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        private void ProcessEndpoints()
```

Comment:
This logic should be dramatically simplified. Azure container apps has all sorts of limitations that cause us to determin what is supported up front. This should be super thin in the docker compose case.